File: #04/q10 - sequencia/sequencia.js
sequencia = (x,y) => x == 0? []
                     : [...Array(x).keys()].map(val => val + y)


console.log(sequencia(0,2));
console.log(sequencia(1,2));
console.log(sequencia(3,5));
console.log(sequencia(4,4));

File: #04/q10 - sequencia/sequencia.hs
sequencia :: (Eq a, Enum a, Num a) => a -> a -> [a]
sequencia 0 b = []
sequencia a b = [b..x]
    where
        x = b + a - 1

main :: IO ()
main = do
    a <- readLn :: IO Int
    b <- readLn :: IO Int
    print $ sequencia a b
